tests/cpp/multidevice.cpp

```diff
@@ -121,11 +121,10 @@ void PipelineTest::validate() {
     if (!output_tv->getDeviceMesh().has(communicator->deviceId())) {
       continue;
     }
-    auto ref_output = isSharded(output_tv) ? shardTensor(
-                                                 ref_unsharded_outputs.at(i),
-                                                 output_tv->getDeviceMesh(),
-                                                 communicator->deviceId())
-                                           : ref_unsharded_outputs.at(i);
+    auto ref_output = isSharded(output_tv)
```

Comment:
With your PR, shardTensor already returns the original tensor when the tv is not sharded. So should we simplify?